tests/pass-dep/libc/libc-epoll.rs

```diff
@@ -650,3 +655,46 @@ fn test_ready_list_fetching_logic() {
     let expected_value1 = fd1 as u64;
     check_epoll_wait::<1>(epfd, &[(expected_event1, expected_value1)]);
 }
+
+// epoll has different behaviour in multithreading. We previously check_and_update_readiness the
+// moment a file description is registered in an epoll instance. But this has an unfortunate side
+// effect of returning notification to another epfd that shouldn't receive notification.
+fn test_epoll_ctl_multithread() {
```

Comment:
The problem doesn't seem related to concurrency at all, so why is this test multithreaded?

Reply:
Right, I discovered this when I am testing it on multi-threaded testcase, but I found a way to write a single-threaded testcase for this.